x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -191,10 +190,37 @@ export function trainedModelsRoutes(
             mlLog.debug(e);
           }
 
-          const body = filterForEnabledFeatureModels(result, getEnabledFeatures());
+          const filteredModels = filterForEnabledFeatureModels(result, getEnabledFeatures());
+
+          try {
+            const jobIdsString = filteredModels.reduce((jobIdsStr, currentModel, idx) => {
+              let id = currentModel.metadata?.analytics_config?.id ?? '';
+              if (id !== '') {
+                id = `${idx > 0 ? ',' : ''}${id}*`;
```

Comment:
This has a bug where if `idx` is > 0 but no dfa jobs have yet to be found it'll add a comma to the front of the string which then causes the loading of dfa jobs to fail.
Rather than looking for `idx > 0`, it should to checking to see if `jobIdsStr` is empty.
I know it was my suggestion to use a `reduce` but I think it was bad advice as it has caused this bug due to the code being hard to read. For the sake of simple to read code I think this should be reverted back to map, filter, join.

```ts
const jobIds = filteredModels
  .map((m) => m.metadata?.analytics_config?.id)
  .filter(isDefined)
  .map((id) => `${id}*`);

if (jobIds.length) {
  const { data_frame_analytics: jobs } = await mlClient.getDataFrameAnalytics({
    id: jobIds.join(','),
    allow_no_match: true,
  });
```

Reply:
Agree - updated in 73216d09bce8ee2503e80b40f20e69817b49e283